browser/lib/util/http.js

```diff
@@ -103,17 +109,8 @@ var Http = (function() {
 		tryAHost(hosts);
 	};
 
-	/**
-	 * Perform an HTTP POST request for a given resolved URI
-	 * @param rest
-	 * @param the full path of the POST request
-	 * @param headers optional hash of headers
-	 * @param body object or buffer containing request body
-	 * @param params optional hash of params
-	 * @param callback (err, response)
-	 */
-	Http.postUri = function(rest, uri, headers, body, params, callback) {
-		Http.Request(rest, uri, headers, params, body, callback || noop);
+	Http.doUri = function(method, rest, uri, headers, body, params, callback) {
+		Http.Request(method, rest, uri, headers, params, body, callback || noop);
```

Comment:
Not a big fan of having both Http.doUri and Http.Request just to swap the order of some parameters. Happy to switch to `do` + `doUri` if you like, so why don't we get rid of `Http.Request` altogether and just make xhrRequest etc. implement `Http.doUri`?